kie-server-parent/kie-server-services/kie-server-services-common/src/main/java/org/kie/server/services/impl/KieServerImpl.java

```diff
@@ -143,7 +143,14 @@ public KieServerImpl() {
                                                                                         containerManager,
                                                                                         this), "KieServer-ControllerConnect");
             connectToControllerThread.start();
-
+            if (Boolean.parseBoolean(currentState.getConfiguration().getConfigItemValue(KieServerConstants.CFG_SYNC_DEPLOYMENT, "true"))) {
+                logger.info("Containers were requested to be deployed synchronously, holding application start...");
+                try {
+                    connectToControllerThread.join();
+                } catch (InterruptedException e1) {
+                    logger.debug("Interrupt exception when waiting for deployments");
```

Comment:
@mswiderski I think you should re-interrupt the thread immediately after catching the InterruptedException otherwise it might be impossible to kill the server because of swallowing the interrupted flag at this point. I tried this but I didn't even see the debug log message so the flag might have been cleared somewhere else but anyway it seems that re-setting the flag in situations like this is the only correct approach, wdyt? http://stackoverflow.com/a/12339487/1691152